modules/src/mock/context.rs

```diff
@@ -196,6 +230,22 @@ impl MockContext {
             ..self
         }
     }
+    pub fn with_connection_capability(
```

Comment:
Maybe? Or have separate `with...` funcs?
```suggestion
    pub fn with_connection_and_port_capability(
```